airflow/jobs/scheduler_job_runner.py

```diff
@@ -1282,9 +1282,15 @@ def _create_dag_runs_dataset_triggered(
                     events=dataset_events,
                 )
 
+                run_conf = {}
+                for event in dataset_events:
+                    if event.extra:
+                        run_conf.update(event.extra)
```

Comment:
I would prefer we put this in a separate PR to discuss. It’s not entirely obvious how extras should be merged from different events that trigger the run.